### PR TITLE
73: Removed setup.py

### LINKED ISSUE #73: Remove setup.py
# Background

- with poetry 1.4.0 it doesn't create the setup.py anymore https://github.com/python-poetry/poetry/releases/tag/1.4.0
- we currently use poetry build to generate the setup.py
- however, the setup.py isn't needed with newer pip versions and if there are releases to pypi or as wheels
- for that reason, we can remove setup.py and githook that generates it, from this repo

# Acceptance Criteria

- [x] Update workflows to poetry 1.4.0
- [x] Remove setup.py Github Workflow
- [x] Remove setup.py githook
- [x] Remove setup.py file

### PR DESCRIPTION
**Pull request by MarleneKress79789**
fixes #73

### CODE REVIEW

File: .github/workflows/check_version.yaml [hunk 1]
```diff
@@ -16,6 +16,6 @@ jobs:
       - name: Install Poetry
         uses: abatilo/actions-poetry@v2
         with:
-          poetry-version: 1.2.0
+          poetry-version: 1.4.0
```
Comment (tkilias) [inline .github/workflows/check_version.yaml]: use dedicated poetry action as in the other projects

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
       - name: Install Poetry
         uses: abatilo/actions-poetry@v2
         with:
-          poetry-version: 1.2.0
```
Comment (tkilias) [inline .github/workflows/release_droid_upload_github_release_assets.yml:24]: use dedicated poetry action as in the other projects

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ jobs:
...
+          poetry-version: 1.4.0
```
Comment (tkilias) [inline .github/workflows/release_droid_upload_github_release_assets.yml]: use dedicated poetry action as in the other projects

File: .github/workflows/test_release.yml [hunk 4]
```diff
@@ -26,7 +26,7 @@ jobs:
     - name: Install Poetry
       uses: abatilo/actions-poetry@v2
       with:
-        poetry-version: 1.2.0
+        poetry-version: 1.4.0
```
Comment (tkilias) [inline .github/workflows/test_release.yml]: use dedicated poetry action as in the other projects

File: .github/workflows/check_ci.yaml [hunk 5]
```diff
@@ -14,7 +14,7 @@ jobs:
         python-version: 3.8
     - uses: abatilo/actions-poetry@v2
       with:
-        poetry-version: 1.2.0
+        poetry-version: 1.4.0
```
Comment (tkilias) [inline .github/workflows/check_ci.yaml]: use dedicated poetry action as in the other projects

File: .github/workflows/check_ci.yaml
Comment (tkilias) [inline .github/workflows/check_ci.yaml:1]: please also fix the job name of this workflow from check_setup_py -> run_unit_tests

File: .github/workflows/check_version.yaml
Comment (tkilias) [inline .github/workflows/check_version.yaml:1]: please disable workflow on main, this can produce issues

**Review by tkilias — CHANGES_REQUESTED** [review]

File: .github/workflows/check_ci.yaml
Comment (MarleneKress79789) [reply .github/workflows/check_ci.yaml:1]: but that is already the case ..

Comment (tkilias) [reply .github/workflows/check_ci.yaml:1]: please check line 6

**Review by tkilias — APPROVED** [review]

**Pull request merged**